### PR TITLE
feat: throw error code in Adyen Provider

### PR DESCRIPTION
**Pull request by Nick-Zag**

### CODE REVIEW

File: packages/demand-bloc-payment/src/providers/adyenErrors.ts [hunk 1]
```diff
@@ -1,5 +1,13 @@
 import { HttpResponseOk, PaymentAuthResponse } from '@karhoo/demand-api'
 
+export class AdyenProviderError extends Error {
```
Comment (aliaksandr-bordak) [inline packages/demand-bloc-payment/src/providers/adyenErrors.ts:3]: For the sake of consistency would be good to name it `AdyenError` like [it is named for Braintree](https://github.com/karhoo/web-lib-demand/blob/master/packages/demand-bloc-payment/src/types.ts#L1)

File: packages/demand-bloc-payment/src/providers/adyenErrors.ts [hunk 2]
```diff
@@ -1,5 +1,13 @@
...
+  code: string
+  constructor(message: string, code: string) {
```
Comment (aliaksandr-bordak) [inline packages/demand-bloc-payment/src/providers/adyenErrors.ts:5]: Would it be more suitable for us to make `code` parameter optional?
`constructor(message: string, code?: string)`

File: packages/demand-bloc-payment/src/providers/adyenErrors.ts [hunk 3]
```diff
@@ -45,14 +53,21 @@ export const handleRefusalResponse = (result: HttpResponseOk<PaymentAuthResponse
...
   }
   return
 }
 
+export const codes = {
+  AE01: 'AE01',
+  AE02: 'AE02',
+  AE03: 'AE03',
+  AE04: 'AE04',
+}
```
Comment (aliaksandr-bordak) [inline packages/demand-bloc-payment/src/providers/adyenErrors.ts:66]: I've noticed that we usually prefer plain JS object to TS enums when dealing with error codes. Are there strong reasons for this approach?

Comment (Nick-Zag) [reply packages/demand-bloc-payment/src/providers/adyenErrors.ts:66]: And what the difference? I think for our purposes it doesn't matter.

Comment (Nick-Zag) [reply packages/demand-bloc-payment/src/providers/adyenErrors.ts:66]: BTW typescript produces additional JS code when compiles enums, and as a result, bigger bundlesize :nerd_face:

**Review by aliaksandr-bordak — APPROVED** [review]

**Pull request merged**